### PR TITLE
mingw fixes

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by neheb** [comment]
interesting... win32 sockets are used under cygwin. Wonder why.

File: include/exiv2/config.h [hunk 1]
```diff
@@ -62,7 +62,7 @@
 
 ///// Path separator macros      /////
 #ifndef EXV_SEPARATOR_STR
-#if defined(WIN32) && !defined(__CYGWIN__)
+#if defined(_WIN32)
```
Comment (kmilos) [inline include/exiv2/config.h:65]: This one might actually be needed? Not 100% sure, but I don't think Cygwin/MSYS understand Windows paths, but they do also define _WIN32 as far as I remember? (potentially applies to all `!defined(__CYGWIN)` occurences below).

Comment (neheb) [reply include/exiv2/config.h:65]: Cygwin does not define _WIN32

Comment (kmilos) [reply include/exiv2/config.h:65]: Indeed - bad memory here, should've checked first...

**Comment by kmilos** [comment]
I guess needs rebase for the conan update now...

**Review by kmilos — APPROVED** [review]

**Pull request merged**